source/patterns/@aws-solutions-constructs/aws-iot-s3/test/iot-s3.test.ts

```diff
@@ -0,0 +1,371 @@
+/**
+ *  Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+import { IotToS3, IotToS3Props } from "../lib";
+import * as cdk from "@aws-cdk/core";
+import '@aws-cdk/assert/jest';
+import * as s3 from "@aws-cdk/aws-s3";
+import { RemovalPolicy } from "@aws-cdk/core";
+
+function deployNewFunc(stack: cdk.Stack) {
+  const props: IotToS3Props = {
+    iotTopicRuleProps: {
+      topicRulePayload: {
+        ruleDisabled: false,
+        description: "process solutions constructs messages",
+        sql: "SELECT * FROM 'solutions/constructs'",
+        actions: []
+      }
+    }
+  };
+  return new IotToS3(stack, 'test-iot-s3-integration', props);
+}
+
+test('check for default props', () => {
+  const stack = new cdk.Stack();
+  const construct = deployNewFunc(stack);
```

Comment:
Why pull this out into a separate function? It just hides the scenario being tested.